lib/rails_deprecation_log_analyser/classifier/equality_comparison_between_parameters_and_hash.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module RailsDeprecationLogAnalyser
+  module Classifier
+    class EqualityComparisonBeweenParametersAndHash < Base
+      def match?(log_line)
+        log_line.include?('Comparing equality between `ActionController::Parameters` and a `Hash` is deprecated and will be removed in Rails 5.1.')
+      end
+
+      protected
+
+      def lines_to_consume
+        1
+      end
+
+      def build_deprecation_warning(lines)
+        warning = DeprecationWarning.new(
+          deprecated: 'Equality comparison between `ActionController::Parameters` and `Hash`',
```

Comment:
The content of the deprecation warning needs to be reviewed. The deprecated line is used as the "type" of the warning for checkstyle. Maybe restrict this to something shorter and we can rely on the message to convey the full warning?